stdlib/public/core/Codable.swift

```diff
@@ -4592,6 +4592,29 @@ extension RawRepresentable where RawValue == Float, Self: Decodable {
   }
 }
 
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, *)
+extension Float16: Codable {
+  /// Creates a new instance by decoding from the given decoder.
+  ///
+  /// This initializer throws an error if reading from the decoder fails, or
+  /// if the data read is corrupted or otherwise invalid.
+  ///
+  /// - Parameter decoder: The decoder to read data from.
+  public init(from decoder: Decoder) throws {
+    self = try Self(Float(from: decoder))
```

Comment:
I'd prefer to be more explicit here about the rounding/truncation behavior.  What happens if someone writes a Float and reads a Float16?

Reply:
I think we should do the same thing as Double -> Float does. I'm not super happy with those semantics, but doing anything different is silly. (So: round in-range values, throw if a finite value is out-of-range)

Reply:
This is done.